### ISSUE TITLE
How to extract feature vectors in a fine-tuned network in keras

### ISSUE THREAD

**Issue by csitaula** [rstudio/keras#331]
I have written a code for pre-trained network and fine tuned with my dataset. I want to extract the features from the layer I added. I am trying to extract the features from layer_dense(units = 1024, activation = 'relu') in my fine tuned model. Could anyone help me to extract the features from that custom layer after fine tuning? Thanks
# create the base pre-trained model
base_model <- application_inception_v3(weights = 'imagenet', include_top = FALSE)

# add our custom layers
predictions <- base_model$output %>% 
  layer_global_average_pooling_2d() %>% 
  layer_dense(units = 1024, activation = 'relu') %>% 
  layer_dense(units = 7, activation = 'softmax')

# this is the model we will train
model <- keras_model(inputs = base_model$input, outputs = predictions)

# first: train only the top layers (which were randomly initialized)
# i.e. freeze all convolutional InceptionV3 layers
freeze_weights(base_model)

# compile the model (should be done *after* setting layers to non-trainable)
model %>% compile(optimizer = 'rmsprop', loss = 'categorical_crossentropy')
train_datagen = image_data_generator(
  rescale = 1/255,
  rotation_range = 40,
  width_shift_range = 0.2,
  height_shift_range = 0.2,
  shear_range = 0.2,
  zoom_range = 0.2,
  horizontal_flip = TRUE,
  fill_mode = "nearest"
)
test_datagen <- image_data_generator(rescale = 1/255)
train_generator <- flow_images_from_directory(
  'd:/nyu/train',
  train_datagen,
  target_size = c(224, 224),
  batch_size = 20,
  class_mode = "sparse"
)
test_generator <- flow_images_from_directory(
  'd:/nyu/test',
  test_datagen,
  target_size = c(224, 224),
  batch_size = 20,
  class_mode = "sparse"
)

model %>% compile(
  loss = "binary_crossentropy",
  optimizer = optimizer_rmsprop(lr = 2e-5),
  metrics = c("accuracy")
)
history <- model %>% fit_generator(
  train_generator,
  steps_per_epoch = 10,
  epochs = 3,
  validation_data = test_generator,
  validation_steps = 5
)
And the link of dataset is:
https://drive.google.com/open?id=10RKbXaUDYvMNtwiIgdVN1h6Kt8oTa1j0
Hope to hear from you soon! Thanks!

**Comment by remya-kannan** [comment]
This should give you an idea of how to go about it:
https://keras.io/getting-started/faq/#how-can-i-obtain-the-output-of-an-intermediate-layer
Hope this helps!
